server/service/src/requisition/program_settings/customer_program_settings/map.rs

```diff
@@ -19,7 +19,18 @@ pub(super) fn map_customer_program_settings(
                 .iter()
                 .filter(|(customer, _)| {
                     customer.program.id == program_setting.program_row.id
-                        || customer.program.elmis_code == program_setting.program_row.elmis_code
+                        || customer
+                            .program
+                            .elmis_code
+                            .as_ref()
+                            .map_or(false, |elmis_code| {
+                                elmis_code
+                                    == program_setting
+                                        .program_row
+                                        .elmis_code
+                                        .as_ref()
+                                        .unwrap_or(&"".to_string())
+                            })
```

Comment:
```suggestion
                        || (customer.program.elmis_code == program_setting.program_row.elmis_code && !customer.program.elmis_code.is_none())
```

Reply:
Can't we just do this?

Was trying to get my head around the filter change as if wasn't entirely clear to me what it was doing.
This seems much clearer and simpler?
Maybe I'm missing something but tests are passing now...

If we need the more complicated filter that's fine but would be nice to have a comment explaining it as it's not immediately obvious, at least to me :)

Reply:
That also looks fine to me! Will test in a bit

Sometimes stores might have different programs that are essentially the same (this is very CIV specific), CIV uses elmis code to match master lists